src/Features/Core/Portable/Completion/Providers/AbstractMemberInsertingCompletionProvider.cs

```diff
@@ -57,14 +58,42 @@ public AbstractMemberInsertingCompletionProvider()
             }
 
             var changes = await newDocument.GetTextChangesAsync(document, cancellationToken).ConfigureAwait(false);
-            return CompletionChange.Create(ImmutableArray.CreateRange(changes), newPosition, includesCommitCharacter: true);
+            var change = Collapse(newText, changes.ToList());
+
+            return CompletionChange.Create(change, newPosition, includesCommitCharacter: true);
+        }
+
+        private TextChange Collapse(SourceText newText, List<TextChange> changes)
```

Comment:
Why is this necessary? Are all providers now expected to return a single TextChange?